### PR TITLE
[ts] Support override modifiers for parameter properties

### LINKED ISSUE #13427: [Bug]: Syntax error is thrown for override modifiers for parameter properties
### 💻

- [X] Would you like to work on a fix?

### How are you using Babel?

Programmatic API (`babel.transform`, `babel.parse`)

### Input code

```ts
class D extends B {
  constructor(protected  override  readonly  foo: string) {
    super(foo);
  }
}
```

### Configuration file name

_No response_

### Configuration

```ts
const { parse } = require("@babel/parser");

parse(`class D extends B {
  constructor(protected  override  readonly  foo: string) {
    super(foo);
  }
}`, { plugins: ["typescript"] });
```

### Current and expected behavior

**Current:**

Syntax error `Unexpected token, expected "," (2:35)` is thrown.([Babel try it out link](https://babeljs.io/repl#?browsers=safari%2012&build=&builtIns=false&corejs=3.6&spec=false&loose=false&code_lz=MYGwhgzhAEAi0FMAeAXBA7AJjAQtA3gFDTTAD26EKATgK7ApnUAUADtWWgwpiWQG4Jq1AJaYEJagjCYKIAJ4kAZmTIAuaFVHoA5gEoCxEptqshzFWT0BuIwF9CDoA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=false&presets=typescript&prettier=false&targets=&version=7.14.4&externalPlugins=))

**Expected:**

Success to parse

### Environment

Use latest `babel/parser` directly.

### Possible solution

_No response_

### Additional context

from: https://github.com/prettier/prettier/issues/11017
[TS Playground Link](https://www.typescriptlang.org/play?#code/MYGwhgzhAEBC0G8BQ1XWAewHYQC4CcBXYXDfACgAd8NcBTEugE2nzrCexAE91sAzAJYBzAFyJoYcXnyCswgNzQARuKyEAtsrr5oAXwCUiFGj1IzSUJBgARaHQAe9LExjxkqTDgLFSFarQM9CzQGABuOrJMdKhsHFy8fFhCYhJS0DJyiipqmtq6hsZoGYSUOuReKQYKJmZmQA)
TypeScript Compiler allows this syntax explicitly (ref: https://github.com/microsoft/TypeScript/pull/43831)

### PR DESCRIPTION
**Pull request by sosukesuzuki**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | Fixes #13427 
| Patch: Bug Fix?          | Y
| Tests Added + Pass?      | Yes
| License                  | MIT

<a href="https://gitpod.io/#https://github.com/babel/babel/pull/13428"></a>

### CODE REVIEW

**Comment by fedeci** [comment]
Do we enforce modifier order in constructor args too?
Because the following seems to be throwing the wrong error:
```ts
class Foo extends Bar {
  constructor(override private readonly foo: string) {
  }
}
```
And can you test this also on a regular class method, because the test should fail in that case as for `readonly`.

File: packages/babel-parser/src/plugins/typescript/index.js [hunk 1]
```diff
@@ -1930,8 +1930,10 @@ export default (superClass: Class<Parser>): Class<Parser> =>
 
       let accessibility: ?N.Accessibility;
       let readonly = false;
+      let override = false;
       if (allowModifiers !== undefined) {
         accessibility = this.parseAccessModifier();
+        override = !!this.tsParseModifier(["override"]);
         readonly = !!this.tsParseModifier(["readonly"]);
```
Comment (JLHwung) [inline packages/babel-parser/src/plugins/typescript/index.js:1937]: We should enforce the modifier order here: `override` must comes before `readonly`.

File: packages/babel-parser/src/plugins/typescript/index.js [hunk 2]
```diff
@@ -1930,8 +1930,10 @@ export default (superClass: Class<Parser>): Class<Parser> =>
...
         if (allowModifiers === false && (accessibility || readonly)) {
```
Comment (JLHwung) [inline packages/babel-parser/src/plugins/typescript/index.js:1938]: should we also check `override` here?

File: packages/babel-parser/src/plugins/typescript/index.js [hunk 3]
```diff
@@ -1930,24 +1931,38 @@ export default (superClass: Class<Parser>): Class<Parser> =>
...
+        this.tsParseModifiers(modified, [
+          "public",
+          "private",
+          "protected",
+          "override",
+          "readonly",
+        ]);
+        accessibility = modified.accessibility;
+        override = modified.override;
+        readonly = modified.readonly;
```
Comment (sosukesuzuki) [inline packages/babel-parser/src/plugins/typescript/index.js:1946]: I used `tsParseModifiers` on an empty object to reuse the logic to enforce the order. What do you think?

**Review by JLHwung — APPROVED** [review]
Thanks!

**Review by fedeci — APPROVED** [review]

**Review by nicolo-ribaudo — APPROVED** [review]
Thanks!

**Pull request merged**